@xen-orchestra/lite/src/components/RemoteConsole.vue

```diff
@@ -9,21 +9,33 @@ import { useXenApiStore } from "@/stores/xen-api.store";
 
 const props = defineProps<{
   location: string;
+  isConsoleAvailable: boolean;
 }>();
 
 const vmConsoleContainer = ref<HTMLDivElement>();
 const xenApiStore = useXenApiStore();
 let vncClient: VncClient | undefined;
 
+const clearVncClient = () => {
+  if (vncClient !== undefined) {
+    vncClient.removeEventListener("disconnect", clearVncClient);
+    if (vncClient._rfbConnectionState !== "disconnected") {
+      vncClient.disconnect();
```

Comment:
As long as you disconnect the VNC here, the event will never be triggered because you are deleting the event listener first. Right?